### DISCUSSION TITLE
Type for resolved child

### DISCUSSION THREAD

**Discussion by captDaylight** [solidjs/solid#1675] [Q&A] [answered]
I'm using a pretty straightforward [bit of code from the documentation](https://www.solidjs.com/docs/latest#children) for `children` but getting this error. Is this a bug?

`child` has the type of `ResolvedJSXElement`

![image](https://user-images.githubusercontent.com/700617/229873888-c1c0f29e-3669-4fc5-9ccc-88821849c885.png)

**Comment by snnsnn** [comment, marked as answer]
This is expected because a child can be of any jsx element and valid jsx elements are not limited to DOM nodes. Since you can not call setAttribute on primitives, you get that error.

**Comment by atk** [comment]
You either need a guard clause against child not being `instanceof HTMLElement` or cast it.
